components/blitz/src/omero/gateway/Gateway.java

```diff
@@ -264,6 +265,8 @@ public ExperimenterData connect(LoginCredentials c)
         } catch (ServerError e) {
             throw new DSOutOfServiceException("Server error", e);
         } catch (SocketException e) {
+            if (e instanceof ConnectionRefusedException)
+                throw new DSOutOfServiceException("Connection refused", e);
             throw new DSOutOfServiceException("Host unreachable", e);
```

Comment:
does `SocketException` happen to already come with a useful message that can just be repeated?